src/app/CommandResponseSender.cpp

```diff
@@ -78,16 +76,33 @@ void CommandResponseSender::OnResponseTimeout(Messaging::ExchangeContext * apExc
 {
     ChipLogDetail(DataManagement, "CommandResponseSender: Timed out waiting for response from requester mState=[%10.10s]",
                   GetStateStr());
+    // We don't need to consider remaining async CommandHandlers here. We become the exchange
+    // delegate only after CommandHandler::OnDone is called.
     Close();
 }
 
-CHIP_ERROR CommandResponseSender::StartSendingCommandResponses()
+void CommandResponseSender::StartSendingCommandResponses()
 {
-    VerifyOrReturnError(mState == State::ReadyForInvokeResponses, CHIP_ERROR_INCORRECT_STATE);
-    // If SendCommandResponse() fails, we are obligated to close the exchange as per the API
-    // contract. However, this method's contract also stipulates that in the event of our
-    // failure, the caller bears the responsibility of closing the exchange.
-    ReturnErrorOnFailure(SendCommandResponse());
+    VerifyOrDie(mState == State::ReadyForInvokeResponses);
+    CHIP_ERROR err = SendCommandResponse();
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DataManagement, "Failed to send InvokeResponseMessage");
+        // TODO(#30453): It should be our responsibility to send a Failure StatusResponse to the requestor
+        // if there is a SessionHandle, but legacy unit tests explicitly check the behavior where
+        // we do not send any message. Changing this behavior should be done in a standalone
+        // PR where only that specific change is made. Here is a possible solution that could
+        // be used that fulfills our responsibility to send a Failure StatusResponse. This causes unit
+        // tests to start failing.
+        //   ```
+        //   if (mExchangeCtx && mExchangeCtx->HasSessionHandle())
+        //   {
+        //       SendStatusResponse(Status::Failure);
+        //   }
+        //   Close();
+        //   return;
+        //   ```
+    }
```

Comment:
but shouldn't we do the `Close(); return;` bit anyway?  It doesn't seem to make sense to run the HasMoreToSend() logic below if we failed to send... unless HasMoreToSend() will always be false in that case?